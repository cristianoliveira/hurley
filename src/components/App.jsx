import React, { Component } from 'react';
import { Link } from 'react-router';
import logo from './../images/logo.svg';

import './../styles/App.css';

import feedcastApi from './../scripts/feedcastApi'
import Search from './Search.jsx'
import PlayerFooter from './PlayerFooter.jsx'

class App extends Component {

  render() {

    return (
      <div className="feedcast">
        <div className="feedcast__header">
          <Link to="/" className="feedcast__logo-wrapper">
            <img src={logo} className="feedcast__logo" alt="logo" />
          </Link>
          <Search/>
        </div>
        <div className="feedcast__navbar">
          <Link activeClassName="active" to="channels"><i className="fa fa-rss"></i> Canais</Link>
          <Link activeClassName="active" to="lastEpisodes"><i className="fa fa-history"></i> Últimos Episódios</Link>
          <Link activeClassName="active" to="categoriesList"><i className="fa fa-th-large"></i> Categorias</Link>
        </div>
        <div className="feedcast__container">
          {this.props.children}
        </div>
        <PlayerFooter />
      </div>
    );
  }
}

export default App;
