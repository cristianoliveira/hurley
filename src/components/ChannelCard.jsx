import React, { Component } from 'react'
import { Link } from 'react-router'

import helpers from './../scripts/helpers'

export default class ChannelCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      imageStatus : false,
      image_url : ''
    }
  }


  componentDidMount() {
    this.checkImage(this.props)
  }


  componentDidUpdate(prevProps) {
    if(this.props.data.image_url !==
        prevProps.data.image_url){
      this.checkImage(this.props)
    }
  }


  checkImage(props){
    let c = props.data
    helpers
      .testImage(c.image_url, 5000)
      .then(e => {
        this.setState({
          imageStatus: true })
      })
      .catch(e => {
        this.setState({
          imageStatus: false })
      })
  }


  thumbnail({image_url}){
    return this.state.imageStatus ? (
      <img
        src={image_url}
        className="feedcast__channel-img"
        onError={()=>this.setState({imageStatus: false})}
      />
    ) : (
      <div className="feedcast__channel-thumbnail">
        <i className="fa fa-podcast"></i>
      </div>
    )
  }

  render () {
    let c = this.props.data
    let thumb = this.thumbnail(c)

    return (
      <Link key={c.uuid} to={`/channel/${c.uuid}`}>
        <div className="feedcast__channel" id={c.uuid}>
          <div className="feedcast__channel-title">
            <h3>{c.title}</h3>
          </div>
          {thumb}
          <div className="feedcast__channel-overlay"></div>
        </div>
      </Link>
    )
  }

}
