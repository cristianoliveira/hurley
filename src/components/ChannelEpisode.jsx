import React, { Component } from 'react'
import helpers from './../scripts/helpers'

class ChannelEpisode extends Component {

  constructor(){
    super();

    this.state = {
      seeMore: false
    }
  }

  render(){

    const { title, summary, audio, description } = this.props.episode
    const descText = summary && summary.length > 0 ? summary : description;

    const summaryEl = descText && descText.length > 0 ? (
      <p className={`feedcast__channelEpisode-summary
        feedcast__channelEpisode-summary--${ this.state.seeMore
          ? 'see-more': 'see-less'}`} >
        {descText}
      </p>
    ):'';

    const btnSeeMore = descText && descText.length > 0 ? (
      <button
        className="feedcast__channelEpisode-btn-see-more"
        onClick={()=> this.setState(
          { seeMore: !this.state.seeMore})}>
        {this.state.seeMore ? 'Ver Menos' : 'Ver Mais'}
      </button>
    ) : '';

    return (
      <div className="feedcast__channelEpisode">
        <div className="feedcast__channelEpisode-button-wrapper">
          <button className="feedcast__channelEpisode-play-button">
            <i className="fa fa-play"></i>
          </button>
        </div>
        <div className="feedcast__channelEpisode-info-wrapper">
          <h4>{title}</h4>
          {summaryEl}
          {btnSeeMore}
          <span>{helpers.secondsToHms(audio.duration)}</span>
        </div>
      </div>
    )
  }
}


export default ChannelEpisode
