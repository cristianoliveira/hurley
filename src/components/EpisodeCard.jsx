import React, { Component } from 'react'
import { connect } from 'react-redux';

import { playEpisode } from 'app/actions/player';
import helpers from './../scripts/helpers'
import { Link } from 'react-router'
import './../styles/EpisodeCard.sass'


class EpisodeCard extends Component {

  playEpisode(){
    this.props.dispatch(
      playEpisode(this.props.episode, this.props.episodes)
    );
  }

  render(){
    const { episode } = this.props
    const background = `url(${episode.channel.image_url})`
    return (
      <div
        className="feedcast__episode-card-wrapper">
        <div className="feedcast__episode-card"
          style={{ background }}
          onClick={() => this.playEpisode()}>
          <span
            className="feedcast__episode-duration">
            {helpers.secondsToHms(episode.audio.duration)}
          </span>
        </div>
        <h3 onClick={() => this.playEpisode()}> {episode.title} </h3>
        <h5>
          <Link to={`/channel/${episode.channel.uuid}/`}>{episode.channel.title}</Link>
        </h5>
      </div>
    )
  }
}

export default connect()(EpisodeCard);
