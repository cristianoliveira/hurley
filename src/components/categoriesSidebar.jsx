import React from 'react';
import { Link } from 'react-router';


import feedcastApi from './../scripts/feedcastApi'


export default class CategoriesSidebar extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      categories : []
    }
  }





  componentDidMount() {
    feedcastApi
    .getCategories()
    .then( data => {
        this.setState({categories : data.categories })
    })
  }




  render(){
    return (
      <div className="feedcast__categories-sidebar">
        { this.state.categories.map((c, i)=>(
          <Link key={i} to={`/category/${c.slug}`}>
            <i className={`fa fa-${c.icon}`}></i> {c.title}
          </Link>)) }
      </div>
    )
  }
}
