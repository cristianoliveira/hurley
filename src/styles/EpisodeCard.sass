@import 'constants'
@import 'mixins'

$font-size: 13px
$line-height: 1.4
$lines-to-show: 2

.feedcast__episode-card-wrapper
  width: $channel-list-size;
  margin-bottom: 20px;

  h3
    cursor: pointer
    font-weight: bold
    color: $darker-blue
    width: 100%
    height: 50px
    display: block
    display: -webkit-box
    height: $font-size*$line-height*$lines-to-show
    text-align: left
    font-size: $font-size
    line-height: $line-height
    -webkit-line-clamp: $lines-to-show
    -webkit-box-orient: vertical
    overflow: hidden
    text-overflow: ellipsis;
    margin-top: 5px
    margin-bottom: 0px

  h5
    margin: 0px
    font-weight: lighter
    a
      text-decoration: none
      &:focus,
      &:visited
        color: #333 !important
      &:hover
        color: $darker-blue !important
        text-decoration: underline !important

.feedcast__episode-card
  cursor: pointer
  position: relative;
  width: 100%;
  height: 150px;
  background-color: white !important;
  background-repeat: no-repeat !important;
  background-size: cover !important;
  background-position: center !important;
  border-radius: 0px;

.feedcast__episode-duration
  position: absolute
  bottom: 0px
  right: 0px
  background-color: rgba(0,0,0,0.8)
  color: white
  padding: 0px 5px
  font-size: 12px
