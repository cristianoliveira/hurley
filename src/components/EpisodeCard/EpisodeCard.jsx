import React, { PureComponent } from 'react'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { playEpisode } from 'app/actions/player';
import helpers from 'app/scripts/helpers'
import { Link } from 'react-router-dom'
import 'app/styles/EpisodeCard.sass'


class EpisodeCard extends PureComponent {

  playEpisode(){
    this.props.dispatch(
      playEpisode(this.props.episode, this.props.episodes.filter(i => i.uuid !== this.props.episode.uuid))
    );
  }

  render(){
    const { episode } = this.props
    const background = `url(${episode.channel.image_url})`
    return (
      <div
        className="feedcast__episode-card-wrapper">
        <div className="feedcast__episode-card"
          style={{ background }}
          onClick={() => this.playEpisode()}>
          <span
            className="feedcast__episode-duration">
            {helpers.secondsToHms(episode.audio.duration)}
          </span>
        </div>
        <h3>
          <Link to={`/${episode.channel.slug}/${episode.slug}/`}>
            {episode.title}
          </Link>
        </h3>
        <h5>
          <Link to={`/${episode.channel.slug}/`}>{episode.channel.title}</Link>
        </h5>
      </div>
    )
  }
}

export default connect()(EpisodeCard);
