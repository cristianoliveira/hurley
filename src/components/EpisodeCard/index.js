import EpisodeCard from './EpisodeCard';
import EpisodeCardList from './EpisodeCardList';

export { EpisodeCardList };
export default EpisodeCard;
