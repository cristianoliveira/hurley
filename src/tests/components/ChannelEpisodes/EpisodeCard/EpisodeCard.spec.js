import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';

import { EpisodeCardList } from 'app/components/EpisodeCard';

describe('EpisodeCardList', () => {
  describe('when has no data', () => {
    it("renders the no episodes", () => {
      const component = renderer.create(<EpisodeCardList />);
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has a list of episodes', () => {
    const state = {
      episodes: [
        {
          uuid: "aadwdadadwawdknaw",
          slug: "pensando-rpg-number-081-entenda-a-figura-do-bruxo-warlock",
          title: "Pensando RPG #081 - Entenda a figura do Bruxo (Warlock)",
          summary: "Neste episódio, falo sobre a figura do Bruxo e a sua respectiva classe em Dungeons & Dragons. Aqui, diferencio-o de outras classes mágicas e dou algumas ideias de jogabilidade e interpretação.",
          description: "Neste episódio, falo sobre a figura do Bruxo e a sua respectiva classe em Dungeons & Dragons. Aqui, diferencio-o de outras classes mágicas e dou algumas ideias de jogabilidade e interpretação.",
          published_at: "2017-08-10T21:58:31+00:00",
          channel: { image_url: 'http://foo/img.svg' },
          audio: {
            url: "http://feeds.soundcloud.com/stream/337354860-user-142845446-pensando-rpg-081-entenda-a-figura-do-bruxo-warlock.mp3",
            size: 12170733,
            duration: 760.633469,
            codec: "mp3",
            bitrate: 128000,
            sample_rate: 44100,
            status: "analysed",
            analysed_at: "2017-08-10T22:17:19+00:00",
            error_message: "",
            error_count: 0
          },
        }
      ]
    }

    it('render the category', () => {
      const component = renderer.create(withStore(<EpisodeCardList {...state} />));
      expect(component).toMatchSnapshot();
    });
  });
})
