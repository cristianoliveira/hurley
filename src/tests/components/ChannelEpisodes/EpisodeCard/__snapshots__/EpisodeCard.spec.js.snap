// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCardList when has a list of episodes render the category 1`] = `
<div
  className="feedcast__episodes-list"
>
  <div
    className="feedcast__episode-card-wrapper"
  >
    <div
      className="feedcast__episode-card"
      onClick={[Function]}
      style={
        Object {
          "background": "url(http://foo/img.svg)",
        }
      }
    >
      <span
        className="feedcast__episode-duration"
      >
        00:12:40
      </span>
    </div>
    <h3>
      <a
        href="/undefined/pensando-rpg-number-081-entenda-a-figura-do-bruxo-warlock/"
        onClick={[Function]}
      >
        Pensando RPG #081 - Entenda a figura do Bruxo (Warlock)
      </a>
    </h3>
    <h5>
      <a
        href="/undefined/"
        onClick={[Function]}
      />
    </h5>
  </div>
</div>
`;

exports[`EpisodeCardList when has no data renders the no episodes 1`] = `null`;
